File: chat-uikit/src/main/java/io/agora/chat/uikit/chathistory/widget/EaseChatRowHistoryText.java
package io.agora.chat.uikit.chathistory.widget;

import android.content.Context;
import android.view.View;
import android.widget.TextView;

import java.util.Date;

import io.agora.chat.ChatMessage;
import io.agora.chat.uikit.R;
import io.agora.chat.uikit.utils.EaseDateUtils;
import io.agora.chat.uikit.widget.chatrow.EaseChatRowText;


public class EaseChatRowHistoryText extends EaseChatRowText {

    public EaseChatRowHistoryText(Context context, boolean isSender) {
		super(context, isSender);
	}

    public EaseChatRowHistoryText(Context context, ChatMessage message, int position, Object adapter) {
		super(context, message, position, adapter);
	}

	@Override
	protected void onInflateView() {
		inflater.inflate(R.layout.ease_row_history_message, this);
	}

	@Override
	public void resetBackground() {

	}

	@Override
	public void updateBackground() {

	}

	@Override
	public void setOtherTimestamp(ChatMessage preMessage) {
		timeStampView.setText(EaseDateUtils.getTimestampString(getContext(), new Date(message.getMsgTime())));
		timeStampView.setVisibility(View.VISIBLE);
	}
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/chat/chathistory/widget/EaseChatRowHistoryText.kt
package io.agora.uikit.feature.chat.chathistory.widget

import android.content.Context
import android.util.AttributeSet
import io.agora.uikit.R
import io.agora.uikit.common.ChatMessage
import io.agora.uikit.common.extensions.getDateFormat
import io.agora.uikit.widget.chatrow.EaseChatRowText

open class EaseChatRowHistoryText @JvmOverloads constructor(
    private val context: Context,
    private val attrs: AttributeSet? = null,
    private val defStyleAttr: Int = 0,
    isSender: Boolean = false
) : EaseChatRowText(context, attrs, defStyleAttr, isSender) {

    override fun onInflateView() {
        inflater.inflate(R.layout.ease_row_history_message, this)
    }

    override fun setOtherTimestamp(preMessage: ChatMessage?) {
        timeStampView?.let {
            preMessage?.let { msg ->
                it.text = msg.getDateFormat(true)
                it.visibility = VISIBLE
            }
        }
    }

    override fun onSetUpView() {
        super.onSetUpView()
        usernickView?.let {
            if (!it.text.toString().trim().isNullOrEmpty()) {
                it.visibility = VISIBLE
            }
        }
    }
}
